Riot/Managers/Call/PiPAnimator.swift

```diff
@@ -0,0 +1,145 @@
+// 
+// Copyright 2020 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+@objc enum PiPAnimationType: Int {
+    case enter
+    case exit
+}
+
+@objcMembers
+class PiPAnimator: NSObject {
+
+    let animationDuration: TimeInterval
+    let animationType: PiPAnimationType
+    weak var pipViewDelegate: PiPViewDelegate?
+    
+    init(animationDuration: TimeInterval,
+         animationType: PiPAnimationType,
+         pipViewDelegate: PiPViewDelegate?) {
+        self.animationDuration = animationDuration
+        self.animationType = animationType
+        self.pipViewDelegate = pipViewDelegate
+        super.init()
+    }
+    
+    private func enterAnimation(context: UIViewControllerContextTransitioning) {
+        guard let keyWindow = UIApplication.shared.keyWindow else {
+            context.completeTransition(false)
+            return
+        }
+        
+        guard let fromVC = context.viewController(forKey: .from) else {
+            context.completeTransition(false)
+            return
+        }
+        
+        fromVC.willMove(toParent: nil)
+        //  TODO: find a way to call this at the end of animation
+        context.completeTransition(true)
+        fromVC.removeFromParent()
+        
+        let pipView = PiPView(frame: fromVC.view.bounds)
+        pipView.contentView = fromVC.view
+        pipView.delegate = pipViewDelegate
+        keyWindow.addSubview(pipView)
+        
+        let transform = CGAffineTransform(scaleX: 0.3, y: 0.3)
```

Comment:
We can store these values with constants at the top of the class

Reply:
👍